packages/node_modules/@webex/internal-plugin-device/test/integration/spec/device.js

```diff
@@ -67,135 +67,68 @@ describe('plugin-device', () => {
-              assert.isTrue(services._getCatalog().status.postauth.collecting);
-              assert.isRejected(device.canRegister(wait));
+          return services.waitForCatalog('postauth')
+            .then(() => {
+              assert.isDefined(services.get('wdm'));
             });
-          });
         });
 
-        describe('when the `wdm` service is not available', () => {
-          let catalog;
-          let services;
-
-          beforeEach('remove wdm service', () => {
-            services = webex.internal.services;
-            /* eslint-disable-next-line no-underscore-dangle */
-            catalog = services._getCatalog();
-
-            catalog.serviceGroups.postauth = [];
-          });
-
-          it('should return a rejected promise', () => {
-            assert.isRejected(device.canRegister(wait));
-          });
+        it('should return a resolved promise', () => {
+          assert.isFulfilled(device.canRegister());
         });
       });
 
-      describe('when wait is falsy', () => {
-        let wait;
-
-        beforeEach('set wait', () => {
-          wait = false;
-        });
-
-        describe('when the `wdm` service is available', () => {
-          let services;
+      describe('when the service catalog is not ready', () => {
+        let services;
 
-          beforeEach('destructure services plugin', () => {
-            services = webex.internal.services;
+        beforeEach('setup catalog to be not ready', () => {
+          services = webex.internal.services;
 
-            return services.updateServices();
-          });
+          services.updateServices();
+        });
 
+        describe('when the \'wdm\' service does exist after wait', () => {
           it('should return a resolved promise', () => {
-            assert.isFulfilled(device.canRegister(wait));
+            /* eslint-disable-next-line no-underscore-dangle */
+            assert.isTrue(services._getCatalog().status.postauth.collecting);
+            assert.isFulfilled(device.canRegister());
           });
         });
 
-        describe('when the service catalog is not ready', () => {
-          let catalog;
-          let services;
-
-          beforeEach('setup catalog to be not ready', () => {
-            services = webex.internal.services;
-            /* eslint-disable-next-line no-underscore-dangle */
-            catalog = services._getCatalog();
-            catalog.serviceGroups.postauth = [];
-
-            services.updateServices();
+        describe('when the \'wdm\' service does not exist after wait', () => {
+          beforeEach('remove wdm service', () => {
+            services.get = sinon.stub().returns(undefined);
+            services.waitForCatalog = sinon.stub().resolves();
           });
 
           it('should return a rejected promise', () => {
             /* eslint-disable-next-line no-underscore-dangle */
             assert.isTrue(services._getCatalog().status.postauth.collecting);
```

Comment:
This assertion isn't described in the test block.

Reply:
implemented.